torchdata/nodes/pin_memory.py

```diff
@@ -115,7 +115,8 @@ def __init__(
         super().__init__()
         self.source = source
         self.snapshot_frequency = snapshot_frequency
-        self._pin_memory = torch.cuda.is_available()
+        self._pin_memory = torch.cuda.is_available() or getattr(torch,
+                                                                torch._C._get_privateuse1_backend_name()).is_available()
```

Comment:
actually, it doesn't look like self._pin_memory is even used anywhere, was this causing an issue for you?

Reply:
Thanks for your reply, I'm trying to test torchtune with torchdata on Ascend device, and find this position not friendly to privateuse1 situations, so I commit this PR :)
I also find that this property is not currently used. I am not sure whether it is a pre-built property or will be abandoned soon?

Reply:
How's your experience with Ascend devices? I haven't tried it before. This property was just copied from torch.utils.data and can likely be dropped

Reply:
> How's your experience with Ascend devices? I haven't tried it before. This property was just copied from torch.utils.data and can likely be dropped

If this property can be dropped, my modification is no longer required, I have changed the content in this pr to remove this property, please help me make sure whether this pr is good to go, thanks :) @andrewkho

Reply:
Thanks @FightingZhen ! If we keep this, we could use it to change this node to just pass-through, though I think I prefer users to handle that themselves so the behaviour is less surprising